crates/polars-plan/src/client/check.rs

```diff
@@ -1,5 +1,69 @@
+use polars_core::error::{polars_err, PolarsResult};
+use polars_io::path_utils::is_cloud_url;
+
 use crate::dsl::Expr;
-use crate::prelude::DslPlan;
+use crate::plans::options::SinkType;
+use crate::plans::{DslFunction, DslPlan, FunctionNode};
+
+/// Assert that the given [`DslPlan`] is eligible to be executed on Polars Cloud.
+pub(super) fn assert_cloud_eligible(dsl: &DslPlan) -> PolarsResult<()> {
+    let mut expr_stack = vec![];
+    for plan_node in dsl.into_iter() {
+        match plan_node {
+            DslPlan::MapFunction {
+                function: DslFunction::FunctionNode(function),
+                ..
+            } => match function {
+                FunctionNode::Opaque { .. } => return ineligible_error("contains opaque function"),
+                #[cfg(feature = "python")]
+                FunctionNode::OpaquePython { .. } => {
+                    return ineligible_error("contains Python function")
+                },
+                _ => (),
+            },
+            #[cfg(feature = "python")]
+            DslPlan::PythonScan { .. } => return ineligible_error("contains Python scan"),
+            DslPlan::GroupBy { apply: Some(_), .. } => {
+                return ineligible_error("contains Python function in group by operation")
+            },
+            DslPlan::Scan { paths, .. }
```

Comment:
We should check for `scan_type: FileScan::Anonymous` as well.

Reply:
Sure - any idea how I could trigger that path from a Python test?

Reply:
Hmm.. I don't think we can atm. A future plugin might though, maybe?

Reply:
All right - I added the branch, it'll just have to go untested for now.